core/src/mindustry/input/Placement.java

```diff
@@ -56,6 +56,30 @@
         return points;
     }
 
+    public static Array<Point2> followLineUntil(int startX, int startY, int endX, int endY){
+        Pools.freeAll(points);
+        points.clear();
+
+        Tile cursor = world.ltile(startX, startY);
+        Block filter = cursor.block();
+
+        while(true){
+            if(cursor.block() != filter) break;                        // keep looking until encountering a different block
+            if(points.contains(Pools.obtain(Point2.class, Point2::new).set(cursor.x, cursor.y))) break; // avoid biting its own tail
```

Comment:
While the original PR did have a change request to make this into a lambda statement, this utilizes a changing `cursor.x, cursor.y` making lambda expression usage error out, I don't have enough knowledge to figure out an alternative way of doing this so any suggestions would be helpful.